rust/automerge/src/iter/list_range.rs

```diff
@@ -57,20 +266,41 @@ impl Shiftable for ListIter<'_> {
 
         let inserts = inserts?.as_usize();
         let pos = self.id.pos() - 1;
-        Some((inserts, action?, value?, id?, pos))
+        Some(List::new(inserts, action?, value?, id?, pos))
+    }
+}
+
+#[derive(Clone, Debug)]
+struct List<'a> {
```

Comment:
As with the `Map` type, this feels like it's not quite the right name as it's not really a list, it's more like `ListOp`, or `ListItemProp`?